Runtime/Interaction/SnapZone.cs

```diff
@@ -428,6 +428,11 @@ public virtual void ForceSelect(XRBaseInteractable interactable)
 
             if (interactable.IsSelectableBy(this))
             {
+                if (interactable is InteractableObject interactableObject)
+                {
+                    interactableObject.ForceSelectEnter(this);
+                }
+                
                 OnSelectEnter(interactable);
```

Comment:
```suggestion
                OnSelectEnter(interactable);
                
                if (interactable is InteractableObject interactableObject)
                {
                    interactableObject.ForceSelectEnter(this);
                }
                
```

Interactor should always do their processes before the interactables.